tests/unit/scripts/test_lint.py

```diff
@@ -533,6 +534,24 @@ def test_invalid_tomlfile(tmp_path):
 )
 def test_parse_checks_reference_mode(ref, expect):
     result = parse_checks(['style'], reference=ref)
-    key = [i for i in result.keys()][-1]
+    key = list(result.keys())[-1]
     value = result[key]
     assert expect in value['short']
+
+
+def test_checks_are_sorted():
+    checks = {}
+    for i in range(2):
+        for origin in [STYLE_CHECKS, UPG_CHECKS]:
+            checks[[k for k in origin][i]] = [v for v in origin.values()][i]
+    sorted_checks = sorted(checks.items(), key=sort_checks)
+    result = [i[1]['purpose'] + str(i[1]['index']) for i in sorted_checks]
+    assert result == ['S1', 'S2', 'U1', 'U2']
```

Comment:
This is a bit gnarly, may I suggest
```suggestion
    sorted_checks = sorted(
        [*UPG_CHECKS.items(), *STYLE_CHECKS.items()],
        key=sort_checks
    )
    assert [
        f"{v['purpose']}{v['index']}" for _, v in sorted_checks
    ] == [
        *[f"S{n}" for n in range(1, len(STYLE_CHECKS) + 1)],
        *[f"U{n}" for n in range(1, len(UPG_CHECKS) + 1)]
    ]
```